evm/src/cpu/kernel/stack/stack_manipulation.rs

```diff
@@ -25,25 +25,50 @@ pub(crate) fn expand_stack_manipulation(body: Vec<Item>) -> Vec<Item> {
     expanded
 }
 
-fn expand(names: Vec<String>, replacements: Vec<StackReplacement>) -> Vec<Item> {
+fn expand(names: Vec<StackPlaceholder>, replacements: Vec<StackReplacement>) -> Vec<Item> {
+    let mut stack_blocks = HashMap::new();
+    let mut stack_names = HashSet::new();
+
     let mut src = names
         .iter()
         .cloned()
-        .map(StackItem::NamedItem)
+        .flat_map(|item| match item {
+            StackPlaceholder::Identifier(name) => {
+                stack_names.insert(name.clone());
```

Comment:
Might be slightly simpler if at this point we treat `Identifier` as a block of size 1? I.e. doing `stack_blocks.insert(name.clone(), 1);` here, and removing `stack_names`. It'd also be a bit easier to check for duplicate names if we only have `stack_blocks`.